### ISSUE TITLE
Ability to pack/localize the shakes, to render on farm/another PC ect..

### ISSUE THREAD

**Issue by loicbramoulle** [EatTheFuture/camera_shakify#27]
Currently if we send a file on another computer or farm, if the addon isn't installed the shakes won't be there.
Could be cool to have a button to pack or localize the actions that contains the shakes, as baking the camera animation is destructive and a bit of a pain to do back & forth as we tweak the files.

**Comment by Griffynbraly** [comment]
wait i had that issue were i couldnt see the shakes when i rendered the my aniamtion with shakfiy how do i fix it?

**Comment by loicbramoulle** [comment]
if you rnedered with another machine that's the issue, shakify isn't or can't be installed there, so you need to bake to action all your cameras, but then the shake is x2 on your PC and shown correctly on render machine that do'nt have the addon.

**Comment by schroef** [comment]
Cant you bake it, thats how its done i believe. Localize is for translation by the way

**Comment by loicbramoulle** [comment]
yea the only way currently is to bake, but that's very expenssive in time, and problematic as it's complex, you need to make sure your base action is set as fake user, properly named, otherwise you would lose it and have to restore it by appendign from a previous verison of the file, if you want to make any change to the shake, taht requires you to bake again ect..

Currently it's extremly simpler to actually extract the shakify data to blank actions, like baking a static camera anim with shakify, and then use the NLA to apply cleanly non-destructively these shakes however you want. But that defeats the purpose of shakify, as then you don't need the addon itself, just extracting the shakes to some action you store somehwere in your library to slap them cleanly on your cameras on any project.
But that's the only clean way to have your shakify shakes work on render farms ect.. (because baking is way too slow and dirty.)

**Comment by schroef** [comment]
@loicbramoulle 

I had a look at the addon. Never used it that much. But indeed i see what mean.  Not even sure how we can bake this method into action. It used constraint and targets an empty which moves in huge numbers

**Comment by schroef** [comment]
But when you render you animation at a renderfarm, you also bake other stuff right. What is so difficult about baking this camera shake. I mean its not impossible.

I just tested it and its works fine, at least my simple test did, i used below settings. 
![settings-to-bake camera](https://github.com/user-attachments/assets/6697c7ff-ffaa-49fb-a9cd-dc72111027dd)

**Comment by loicbramoulle** [comment]
The issue is that sometimes it takes up to 10 minutes to bake one cam, because you have various things in your scene, and thousands of frames.
And if you have 15 cameras then it can take an hour, and you still have to make sure your original camera action is backed up as fake user, properly named, and you get 2X shake locally unless you set your shakes to zero, but then you do'nt know what were their value. it's basically a lot of potential sources of pain, either it just takes raw time, or just creates errors that you will need time to fix ect... (finding a camera action backup in an older file as you forgot fake user ect...)

Just my impression but if the shakes were applied via the NLA, it would all be blender vanilla by default, but maybe that's totally different than what shakify does currently.

**Comment by schroef** [comment]
Yeah 8 guess hea t scene will take time indeed. I just wonder if a render farm allows a user to run a script at runtime. That could fix the issue as @cessen suggested.

We could easily make the custom operator backup actions coree tly before baking. The settings I showed will bake everything properly and also clear the constraints, so you don't end up with double shake settings.

Ps isn't proper workflow to save a custom versions of your file file render and called perhaps _RENDER or _BAKED. That way you always have the original file.

The bake operator could also automatically go over all cameras with shakify active. Isn't that difficult to code

**Comment by cessen** [comment]
I've added a new feature in [Camera Shakify v0.4.0](https://extensions.blender.org/add-ons/camera-shakify/manage/versions/#v040) that adds the needed Python script to the blend file with a single button press.  The button is labeled "Prep Blend File For Render Farm", and is under the "Misc Utilities" section of the Camera Shakify panel.

Can you guys give it a test and see if it works for the farms you're using?

**Comment by loicbramoulle** [comment]
Tested on OCTA.render and it works perfectly !
Thanks again !!

**Comment by cessen** [comment]
Thanks for reporting back!  I'll close this issue then.
